core/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java

```diff
@@ -0,0 +1,479 @@
+                .build();
+        IndexMetaData metaData = IndexMetaData.builder("_na_").settings(build).build();
+        this.notAvailableIndexSettings = new IndexSettings(metaData, Settings.EMPTY, Collections.emptyList());
+
+        this.tokenFilters = Collections.unmodifiableMap(tokenFilterBuilder);
+        this.tokenizers = Collections.unmodifiableMap(tokenizerBuilder);
+        this.charFilters = Collections.unmodifiableMap(charFilterBuilder);
+        this.analyzers = Collections.unmodifiableMap(analyzerBuilder);
+    }
+
+    /**
+     * Returns a registered {@link TokenizerFactory} provider by name or <code>null</code> if the tokenizer was not registered
+     */
+    public AnalysisModule.AnalysisProvider<TokenizerFactory> getTokenizerProvider(String tokenizer) {
+        return tokenizers.getOrDefault(tokenizer, this.prebuildAnalysis.getTokenizerFactory(tokenizer));
+    }
+
+    /**
+     * Returns a registered {@link TokenFilterFactory} provider by name or <code>null</code> if the token filter was not registered
+     */
+    public AnalysisModule.AnalysisProvider<TokenFilterFactory> getTokenFilterProvider(String tokenFilter) {
+        return tokenFilters.getOrDefault(tokenFilter, this.prebuildAnalysis.getTokenFilterFactory(tokenFilter));
+    }
+
+    /**
+     * Returns a registered {@link CharFilterFactory} provider by name or <code>null</code> if the char filter was not registered
+     */
+    public AnalysisModule.AnalysisProvider<CharFilterFactory> getCharFilterProvider(String charFilter) {
+        return charFilters.getOrDefault(charFilter, this.prebuildAnalysis.getCharFilterFactory(charFilter));
+    }
+
+    /**
+     * Returns a registered {@link Analyzer} provider by name or <code>null</code> if the analyzer was not registered
+     */
+    public Analyzer getAnalyzer(String analyzer) throws IOException {
+        AnalysisModule.AnalysisProvider<AnalyzerProvider> analyzerProvider = this.prebuildAnalysis.getAnalyzerProvider(analyzer);
+        if (analyzerProvider == null) {
+            AnalysisModule.AnalysisProvider<AnalyzerProvider> provider = analyzers.get(analyzer);
+            return provider == null ? null : cachedAnalyzer.computeIfAbsent(analyzer, (key) -> {
+                        try {
+                            return provider.get(notAvailableIndexSettings, environemnt, key, DEFAULT_SETTINGS).get();
+                        } catch (IOException ex) {
+                            throw new ElasticsearchException("failed to load analyzer for name " + key, ex);
+                        }}
+            );
+        }
+        return analyzerProvider.get(notAvailableIndexSettings, environemnt, analyzer, DEFAULT_SETTINGS).get();
+    }
+
+    /**
+     * A dummy index settings object to create instances from {@link org.elasticsearch.indices.analysis.AnalysisModule.AnalysisProvider} outside of
+     * the scope of an index.
+     */
+    public IndexSettings getDummyIndexSettings() {
```

Comment:
Can this be private? Does it really need an accessor if it is just used internally within getAnalyzer?